### ISSUE TITLE
RFC 7518 specifies SHA lengths?

### ISSUE THREAD

**Issue by FGasper** [DCIT/perl-CryptX#45]
https://tools.ietf.org/html/rfc7518#page-8

^^ There doesn’t seem to be any use given for, say, P-256 with SHA-1, though that is the default for the RFC 7518 methods in Crypt::PK::ECC.

Should the defaults be changed to correspond to the curve length? Otherwise each implementation will need to maintain the mapping on the page linked above.

**Comment by karel-m** [comment]
I do not think this should be handled at `Crypt::PK::ECC` level. In fact I already have this sort "mapping" in another module https://metacpan.org/source/MIK/Crypt-JWT-0.021/lib/Crypt/JWT.pm#L521

The `rfc7518` in `sign_message_rfc7518` is more about the output format of the signature.

**Comment by FGasper** [comment]
What is the intended use of being able to sign things with `sign_message_rfc7518` in a way that contradicts RFC 7518, such as a P-256 key with SHA-1?

For example, if I have a P-256 key, the hash algorithm that the RFC describes is SHA-256, and only SHA-256.

**Comment by FGasper** [comment]
OK, so the function name is meant to be more “sign in the style of RFC 7518” rather than “sign according to RFC 7518”.

FWIW, for me, anyway, yeah, it would be helpful if the documentation were like:

> NOTE: This creates/verifies signatures according to the structure that RFC 7518 but does not necessarily follow the hashing algorithm as the RFC describes. You’ll still need to specify, e.g., `sha256` for a P-256 key.

I’m still curious what application is served by putting SHA-1 with an RFC 7518 signature, but anyway. Thank you for your time and for maintaining this very useful module. :)
